lips/lip-1.md

```diff
@@ -1,9 +1,444 @@
+| _ObjectType| str| Y | Fixed value: `CommandRequestObject`|
+|command_type | str| Y |A string representing the type of command contained in the request. |
+| command | Command object | Y | The command to sequence. |
+
+<details>
+<summary> CommandRequestObject example </summary>
+<pre>
+{
+    "_ObjectType": "CommandRequestObject",
+    "command_type": CommandType,
+    "command": CommandObject(),
+}
+</pre>
+</details>
+
+### CommandResponseObject
+All responses to a CommandRequestObject are in the form of a CommandResponseObject
+
+| Field 	     | Type 	| Required? 	| Description 	|
+|-------	     |------	|-----------	|-------------	|
+| _ObjectType    | str      | Y             | The fixed string `CommandResponseObject`. |
+| status         | str      | Y             | Either `success` or `failure`. |
+| error          | List of [OffChainErrorObject](#offchainerrorobject) | N | Details on errors when status == "failure"
+
+<details>
+<summary> CommandResponseObject example </summary>
+<pre>
+{
+    "_ObjectType": "CommandResponseObject",
+    "error": [OffChainErrorObject()],
+    "status": "failure"
+}
+</pre>
+</details>
+
+When the `CommandResponseObject` status field is `failure`, the `error` field is included in the response to indicate the nature of the failure. The `error` field (type `OffChainError`) is a list of OffChainError objects. 
+
+### OffChainErrorObject
+Represents an error that occurred in response to a command.
+
+| Field 	     | Type 	| Required? 	| Description 	|
+|-------	     |------	|-----------	|-------------	|
+| type    | str (enum)     | Y             | Either "command_error" or "protocol_error" |
+| field            | str       | N             | The field on which this error occurred|
+| code    | str (enum) | Y    | The error code of the corresponding error |
+| message         | str      | N             | Additional details about this error |
+
+<details>
+<summary> OffChainErrorObject example </summary>
+<pre>
+{
+    "type": "command_error",
+    "field": "0.sender.kyc_data.surname",
+    "code": "missing_data",
+    "message": "",
+}
+
+</pre>
+</details>
+
+# Command Sequencing
+
+The low-level Off-Chain protocol allows two VASPs to sequence request-responses for commands originating from either VASP, in order to maintain a consistent database of shared objects. Sequencing a command requires both VAPSs to confirm it is valid, as well as its sequence in relation to other commands operating upon the same objects.  Since commands may operate upon multiple objects, a command only succeeds if the command is able to be applied to every dependent object - ensuring atomicity of the command and consistency of the objects. Both VASPs in a channel can asynchronously attempt to initiate and execute commands on shared objects. All commands upon shared objects which are exchanged between pairs of VASPs are sequenced relative to the prior state of each shared object in the command.
```

Comment:
there is no mention of "shared object" until here. can you clarify how request/response/error objects are related to "shared objects" and why any objects are "shared"?

Reply:
Updated it in the latest commit.  There is one earlier reference to shared objects, but it didn't describe them well, so added a better description there